js/app.js

```diff
@@ -28,74 +31,102 @@ alert(`Oh! ${userName}! I didn't recognize you. Come on in.`);
 alert('Let\'s play a game. Answer some trivia about me. Go for a high score! Or a low score if you\'re too afraid of success. Acceptable answers are in the form yes/no and y/n, unless otherwise noted.');
 
-//Initialize a user answer
-let userAnswer = '';
+quizGame();
+
+//This function returns a random integer between min and max (both included)
+
+function randomInteger(min, max){
+  return Math.floor(Math.random() * (max - min + 1 )) + min;
+
+}
+
+function quizGame (){
+  // Loop through the questions
+  for (let i = 0; i < numberOfQuestions; i++) {
+
+    //For readability, define a question number
+    let questionNumber = i + 1;
 
-// Loop through the questions
-for (let i = 0; i < numberOfQuestions; i++) {
+    // For questions 1 - 5, a straight yes/no or y/n is checked
+    if (questionNumber <= 5) {
+
+      yesNoQuestions(questionNumber, questionsArray[i], yesNoAnswersArray[i]);
+
+      // For question 6, give the user 4 chances at guessing the random number.
+    } else if (questionNumber === 6) {
+
+      ranNumQuestion(questionNumber, questionsArray[i], randomNumberAnswer, 4);
+
+      // For question 7, give the user 6 attempts at guessing correctly.
+    } else {
+
+      travelQuestion(questionNumber, questionsArray[i], topThreeTravelAnswers, 6);
+
+    }
+  }
+  showScore();
+}
 
-  //For readability, define a question number
-  let questionNumber = i + 1;
+function yesNoQuestions (questionNumber, question, answer) {
+  let userAnswer = prompt(question).toLowerCase();
 
-  // For questions 1 - 5, a straight yes/no or y/n is checked
-  if (questionNumber <= 5) {
-    userAnswer = prompt(questionsArray[i]).toLowerCase();
-    if (userAnswer === yesNoAnswersArray[i] || userAnswer === yesNoAnswersArray[i].charAt(0)) {
+  if (userAnswer === answer || userAnswer === answer.charAt(0)) {
     // console.log('Question' + (questionNumber) + ": Correct!");
+    alert(`Question ${questionNumber}: Correct!`);
+    scoreCount++;
+  } else {
+    // console.log('Question' + (questionNumber) + ': Wrong stupid!')
+    alert(`Question ${questionNumber}: Wrong, stupid!`);
+  }
+}
+
+function ranNumQuestion(questionNumber, question, answer, attempts){
+  let attemptsLeft = attempts;
+  for (let i = 0; i < attempts; i++) {
+    let userAnswer = prompt(question);
+    attemptsLeft--;
+    if (userAnswer < answer) {
+      alert(`Too low. You have ${attemptsLeft} attempts left.`);
+    } else if (userAnswer > answer) {
+      alert(`Too high. You have ${attemptsLeft} attempts left.`);
+    } else {
       alert(`Question ${questionNumber}: Correct!`);
       scoreCount++;
-    } else {
-    // console.log('Question' + (questionNumber) + ': Wrong stupid!')
-      alert(`Question ${questionNumber}: Wrong, stupid!`);
+      break;
     }
+    if (attemptsLeft === 0) {
+      alert(`Question ${questionNumber}: Incorrect. Pitiful. Even with a generous number of attempts you couldn't deduce that the answer was ${answer}.`);
+      break;
```

Comment:
This break statement isn't necessary, since when attempts === 0, you should be on your last iteration of the loop.